sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/CustomModelTransforms.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.ai.formrecognizer;
+
+import com.azure.ai.formrecognizer.implementation.models.ErrorInformation;
+import com.azure.ai.formrecognizer.implementation.models.Model;
+import com.azure.ai.formrecognizer.implementation.models.ModelInfo;
+import com.azure.ai.formrecognizer.implementation.models.ModelStatus;
+import com.azure.ai.formrecognizer.models.CustomFormModel;
+import com.azure.ai.formrecognizer.models.CustomFormModelField;
+import com.azure.ai.formrecognizer.models.CustomFormSubModel;
+import com.azure.ai.formrecognizer.models.FormRecognizerError;
+import com.azure.ai.formrecognizer.models.ModelTrainingStatus;
+import com.azure.ai.formrecognizer.models.TrainingDocumentInfo;
+import com.azure.ai.formrecognizer.models.TrainingStatus;
+import com.azure.core.util.CoreUtils;
+import com.azure.core.util.IterableStream;
+import com.azure.core.util.logging.ClientLogger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+
+/**
+ * Helper class to convert service level custom form related models to SDK exposed models.
+ */
+final class CustomModelTransforms {
+    private static final ClientLogger LOGGER = new ClientLogger(CustomModelTransforms.class);
+
+    private CustomModelTransforms() {
+    }
+
+    /**
+     * Helper method to convert the {@link Model model Response} from service to {@link CustomFormModel}.
+     *
+     * @param modelResponse The {@code Model model response} returned from the service.
+     *
+     * @return The {@link CustomFormModel}.
+     */
+    static CustomFormModel toCustomFormModel(Model modelResponse) {
+        ModelInfo modelInfo = modelResponse.getModelInfo();
+        if (modelInfo.getStatus() == ModelStatus.INVALID) {
+            throw LOGGER.logExceptionAsError(
+                new IllegalArgumentException(String.format("Model Id %s returned with status: %s",
+                    modelInfo.getModelId(), modelInfo.getStatus())));
+        }
+
+        List<TrainingDocumentInfo> trainingDocumentInfoList =
+            modelResponse.getTrainResult().getTrainingDocuments().stream()
```

Comment:
not sure how Java does it, but asking the same Krista pointed out in in .net,
TrainResult can be null. So, will `modelResponse.getTrainResult().getTrainingDocuments()` and all other references work?

Reply:
Updated to check for trainResult as it can be null before referencing the training documents.